utam-compiler/src/main/java/utam/compiler/types/BasicElementInterface.java

```diff
@@ -135,7 +135,7 @@ public static boolean isBasicType(TypeProvider type) {
       return values.toArray(String[]::new);
     }
     if(isThrowAnError) {
-      throw new UtamCompilationError(typeNodeValueError);
+      throw new IllegalArgumentException(typeNodeValueError);
```

Comment:
Why the exception type change? Not saying it's the wrong thing to do, just curious.

Reply:
refactored to throw compiler error